src/gui.rs

```diff
@@ -165,25 +153,46 @@ impl Application for State {
             Command::batch([
                 Command::perform(fetch_minecraft_versions(), Message::SetMcVersions),
                 Command::perform(fetch_loader_versions(), Message::SetLoaderVersions),
-            ])
+            ]),
         )
     }
 
     fn title(&self) -> String {
-        "Quilt Installer".to_owned()
+        format!(
+            "Quilt Installer{}",
+            if self.is_installing {
+                " - Installing"
+            } else {
+                ""
+            }
+        )
     }
 
-    fn update(&mut self, message: Self::Message, _clipboard: &mut Clipboard) -> Command<Self::Message> {
+    fn update(&mut self, message: Self::Message) -> Command<Self::Message> {
         match message {
             Message::Interaction(interaction) => match interaction {
                 Interaction::ChangeClientLocation(location) => self.client_location = location,
                 Interaction::BrowseClientLocation => return Message::BrowseClientLocation.into(),
                 Interaction::Install => return Message::Install.into(),
-                Interaction::SelectInstallation(installation) => self.selected_installation = installation,
-                Interaction::SelectLoaderVersion(version) => self.selected_loader_version = Some(version),
-                Interaction::SelectMcVersion(version) => self.selected_minecraft_version = Some(version),
-                Interaction::EnableSnapshots(enable) => self.show_snapshots = enable,
-                Interaction::EnableBetas(enable) => self.show_betas = enable,
+                Interaction::SelectInstallation(i) => self.installation_type = i,
+                Interaction::SelectLoaderVersion(v) => self.selected_loader_version = Some(v),
+                Interaction::SelectMcVersion(v) => self.selected_minecraft_version = Some(v),
+                Interaction::ToggleSnapshots(enable) => {
+                    self.show_snapshots = enable;
+                    self.selected_minecraft_version = self
+                        .minecraft_versions
+                        .iter()
+                        .find(|v| enable || v.stable)
+                        .cloned();
+                }
+                Interaction::ToggleBetas(enable) => {
+                    self.show_betas = enable;
+                    self.selected_loader_version = self
+                        .loader_versions
+                        .iter()
+                        .find(|v| enable || !v.version.pre.contains("beta"))
```

Comment:
Something that I've done on my branch: `beta` is unreliable once alphas appear; I'd consider checking for a dash instead, since that is always used for semver's pre-release mechanism